rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -7609,6 +7609,134 @@ public void getQueuedPegoutsCount_after_RSKIP271_activation() throws IOException
         assertEquals(2, bridgeSupport.getQueuedPegoutsCount());
     }
 
+    @Test
+    public void getEstimatedFeesForNextPegOutEvent_before_RSKIP271_activation() throws IOException {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP271)).thenReturn(false);
+
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withActivations(activations)
+                .withProvider(provider)
+                .build();
+
+        verify(provider, never()).getFeePerKb();
+        assertEquals(Coin.ZERO, bridgeSupport.getEstimatedFeesForNextPegOutEvent());
+    }
+
+    @Test
+    public void getEstimatedFeesForNextPegOutEvent_after_RSKIP271_activation() throws IOException {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP271)).thenReturn(true);
+
+        int pegoutRequestsCount = 5;
+        Coin feePerKB = Coin.MILLICOIN;
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+        when(provider.getReleaseRequestQueueSize()).thenReturn(pegoutRequestsCount);
+        when(provider.getFeePerKb()).thenReturn(feePerKB);
+
+        Federation federation = bridgeConstants.getGenesisFederation();
+        FederationSupport federationSupport = mock(FederationSupport.class);
+        when(federationSupport.getActiveFederation()).thenReturn(federation);
+
+        BridgeSupport bridgeSupport = new BridgeSupport(
+                bridgeConstants,
+                provider,
+                mock(BridgeEventLogger.class),
+                new BtcLockSenderProvider(),
+                new PeginInstructionsProvider(),
+                mock(Repository.class),
+                mock(Block.class),
+                mock(Context.class),
+                federationSupport,
+                mock(BtcBlockStoreWithCache.Factory.class),
+                activations
+        );
```

Comment:
You can simply use BridgeSupportBuilder class here

Reply:
Using BridgeSupportBuilder Doesn't Instantiate Federation Support For Some Weird Reasons

Reply:
Resolved